### DISCUSSION TITLE
Update to Security Onion 2.3.60 breaks Elastalert

### DISCUSSION THREAD

**Discussion by pkgwillie** [Security-Onion-Solutions/securityonion#4815] [Unsupported Versions]
After updating to Security Onion 2.3.60, ElastAlert shows as missing in so-status

How do I fix this?
![so-status](https://user-images.githubusercontent.com/7101154/125771590-5489f648-932b-462b-95c7-3fb48b8d0cea.JPG)

Dashboard
![so-dashboard](https://user-images.githubusercontent.com/7101154/125773743-4ec3d1e2-55af-47d4-85dd-8d2d2d7c4dc4.JPG)

**Comment by pkgwillie** [comment]
Is there anyway to force **soup**  to reinstall to fix missing container ? I have searched through documentation with no luck.

**Reply by pkgwillie** [reply]
Fix: 
Turns out that elastalert config was pointing to wrong rules folder.
Anyone with same issue should update the elastalert config **/opt/so/conf/elastalert/elastalert_config.yaml**

Line: 
rules_folder: /opt/elastalert/rules/

should be updated to 
**/opt/so/rules/elastalert**


![so-status_1](https://user-images.githubusercontent.com/7101154/125885725-0fa1e5d2-33f6-4cf2-97a4-a8142fdc32be.JPG)

**Reply by petiepooo** [reply]
That config file is loaded by the application running in docker, so the paths specified there are not the same as the paths available within the container.  If you run 'sudo docker exec -i so-elastalert ls /opt/elastalert/rules' you will see the playbook directory.  This is because /opt/so/rules/elastalert is volume-mounted (docker bind) to opt/elastalert/rules in the container.

I recommend you change that setting back and continue to debug, @pkgwillie.  I had some trouble getting elastalert to run after upgrading to 2.3.60, but the troubles resolved over time.  I'm not sure what the root cause was.  Logfiles for elastalert are in /opt/so/log/elastalert.

**Reply by pkgwillie** [reply]
@petiepooo thanks for the insight. Will reverse this and see if it resolves by itself.

**Reply by pkgwillie** [reply]
@petiepooo  upon checking config I see that change I made had been reversed. So am good, since ElastAlert is still running.

**Reply by bicfordio** [reply]
Ran the 2.3.60 update last night, getting the "missing" error for a few different docker containers at this point. Are you saying this just eventually resolves itself after awhile?

**Reply by bicfordio** [reply]
My containers are missing for so-soc, so-elastalert, and so-elasticsearch. When I try to run a so-xxx-start command it comes back with this error, which seems to point towards somewhere having an odd IP address configured incorrectly. I get the 502 bad gateway error by ngnix, so wondering if there is a poor network assignment in the back end somewhere

![Screen Shot 2021-07-16 at 1 50 29 PM](https://user-images.githubusercontent.com/5385093/125995234-cc88d7a9-3128-4cf0-8fa5-02166290cc3b.png)

**Reply by pkgwillie** [reply]
@bicfordio for the missing container errors I will advice that you give it time like @bicfordio  suggested. Sounds counterintuitive but actually works. For the IP I suggest you check on any typos in the IP address.
